### PR TITLE
Fix Indentation of Code Block in several pages

### PR DESCRIPTION
**Pull request by VoxelMC**
Note: `I would like this PR to be considered for Hacktoberfest`

#### What kind of changes does this PR include?

- Minor content fixes (broken links, typos, etc.)

#### Description

Corrected indentation for each language on the [the middleware documentation page.](https://docs.astro.build/en/guides/middleware/)

**This pull request has been recreated from #4583 without the changes to i18n pages.**

image

To

image

### CODE REVIEW

**Review by dreyfus92 — APPROVED** [review]
Thank you for this @VoxelMC 😁, LGTM ✅

**Review by Genteure — CHANGES_REQUESTED** [review]
I think indentation should be added to the code block under `3.`, instead of removing the indentation before the code block in `2.`.

The code block should be a child element of the ordered list.

------

Source text:

``````markdown
1. one
2. two
    ```js
    console.log(1)
    ```
3. three
    ```js
    console.log(1)
    ```
``````

Rendered result:

1. one
2. two
    ```js
    console.log(2)
    ```
3. three
    ```js
    console.log(3)
    ```

The html looks like:

```html
<ol>
  <li>one</li>
  <li>two <code>console.log(2)</code></li>
  <li>three <code>console.log(3)</code></li>
</ol>
```

------

Compared to:

``````markdown
1. one
2. two
```js
console.log(1)
```
3. three
```js
console.log(1)
```
``````

Rendered result:

1. one
2. two
```js
console.log(2)
```
3. three
```js
console.log(3)
```

The html looks like:

```html
<ol>
  <li>one</li>
  <li>two</li>
</ol>
<code>console.log(2)</code>
<ol>
  <li>three</li>
</ol>
<code>console.log(3)</code>
```

**Comment by VoxelMC** [comment]
> I think indentation should be added to the code block under `3.`, instead of removing the indentation before the code block in `2.`.
> 
> The code block should be a child element of the ordered list.
> 
> Source text:
> 
> ```
> 1. one
> 2. two
>     ```js
>     console.log(1)
>     ```
> 3. three
>     ```js
>     console.log(1)
>     ```
> ```
> 
> Rendered result:
> 
> 1. one
> 2. two
>    ```js
>    console.log(2)
>    ```
> 3. three
>    ```js
>    console.log(3)
>    ```
> 
> The html looks like:
> 
> ```
> <ol>
>   <li>one</li>
>   <li>two <code>console.log(2)</code></li>
>   <li>three <code>console.log(3)</code></li>
> </ol>
> ```
> 
> Compared to:
> 
> ```
> 1. one
> 2. two
> ```js
> console.log(1)
> ```
> 3. three
> ```js
> console.log(1)
> ```
> ```
> 
> Rendered result:
> 
> 1. one
> 2. two
> 
> ```js
> console.log(2)
> ```
> 
> 3. three
> 
> ```js
> console.log(3)
> ```
> 
> The html looks like:
> 
> ```
> <ol>
>   <li>one</li>
>   <li>two</li>
> </ol>
> <code>console.log(2)</code>
> <ol>
>   <li>three</li>
> </ol>
> <code>console.log(3)</code>
> ```

My personal opinion is that this approach is inconsistent with the overall aesthetic of the pages. Overall, it varies from page to page in the documentation. I am just aiming for consistency in terms of the left margin. 

However, I am more than willing to make the changes if some more people agree!

**Comment by sarah11918** [comment]
Hey Hey! I'm here to confirm that @Genteure is correct with what we *intend* to use in docs! Extra indentation in these numbered lists!

It's a pain and we don't always get it right, but when we notice, we always do correct this way. So if you're happy to re-indent everything, @VoxelMC , that would be fabulous! 🙌 ➡️

**Comment by VoxelMC** [comment]
> Hey Hey! I'm here to confirm that @Genteure is correct with what we _intend_ to use in docs! Extra indentation in these numbered lists!
> 
> It's a pain and we don't always get it right, but when we notice, we always do correct this way. So if you're happy to re-indent everything, @VoxelMC , that would be fabulous! 🙌 ➡️

Yes! I can definitely do that :)

Watch out, inconsistent indentation!

**Comment by sarah11918** [comment]
@VoxelMC are you still interested in saving us from our indentation??  :smile:

**Comment by VoxelMC** [comment]
> @VoxelMC are you still interested in saving us from our indentation??  :smile: 

Yup! I will get started on that soon! Just had a project deadline yesterday... :p

**Comment by VoxelMC** [comment]
@sarah11918 I believe I have found all of the culprits! There is one on the Cloudflare README, so I may open up a PR there too, for consistency's sake. Let me know if it's okay!

**Comment by sarah11918** [comment]
Hi @VoxelMC ! Just to let you know, in future I would really appreciate if these were separate PRs. This is 18 files to go through and check previews, some with multiple changes at different points on a page, and it is now much bigger than the original PR.

I appreciate the enthusiasm, and obviously we do want these all fixed! Just know that these changes turn a very simple PR, that can be merged quite quickly, into one that requires a lot of checking on the part of the reviewers. For example, this is not something I can approve in 30 seconds anymore when I have a spare minute. Now, I have to keep a checklist to make sure I haven't missed verifying any of the changes!

I will edit this one as is, but in future please do not do this as it is not actually as helpful as you might think it is. 😅  A handy rule is, if your changes no longer reflect the original PR title, you may have gotten a little *too* enthusiastic in your changes...

**Comment by VoxelMC** [comment]
> Hi @VoxelMC ! Just to let you know, in future I would really appreciate if these were separate PRs. This is 18 files to go through and check previews, some with multiple changes at different points on a page, and it is now much bigger than the original PR.
> 
> I appreciate the enthusiasm, and obviously we do want these all fixed! Just know that these changes turn a very simple PR, that can be merged quite quickly, into one that requires a lot of checking on the part of the reviewers. For example, this is not something I can approve in 30 seconds anymore when I have a spare minute. Now, I have to keep a checklist to make sure I haven't missed verifying any of the changes!
> 
> I will edit this one as is, but in future please do not do this as it is not actually as helpful as you might think it is. 😅  A handy rule is, if your changes no longer reflect the original PR title, you may have gotten a little *too* enthusiastic in your changes...

Yes of course, I am sorry! I won't do it again. I realize my overstepping. I won't do it again, and from now on I will keep the perspective of the reviewers at the top of my list. I know you have a lot of work to do.

File: src/content/docs/en/tutorial/3-components/4.mdx [hunk 1]
```diff
@@ -46,86 +46,86 @@ Create a `<Hamburger />` component to open and close your mobile menu.
...
+      padding: 10px 0;
+      text-decoration: none;
+      font-size: 1.2rem;
+      font-weight: bold;
+      text-transform: uppercase;
     }
 
-    .hamburger {
-      display: none;
+    .nav-links a:hover, a:focus {
```
Comment (sarah11918) [inline src/content/docs/en/tutorial/3-components/4.mdx:102]: This is another reason not to include several files in one PR. It took me several minutes of closely inspecting here to figure out why this is showing as a change, and whether or not something had actually changed by mistake.

File: src/content/docs/en/recipes/add-yaml-support.mdx [hunk 2]
```diff
@@ -4,16 +4,30 @@ description: Learn how you can import YAML data by adding a Rollup plugin to you
...
 Astro builds on top of Vite, and supports both Vite and Rollup plugins. This recipe uses a Rollup plugin to add the ability to import a YAML (`.yml`) file in Astro.
 
 ## Recipe
 
 1. Install `@rollup/plugin-yaml`:
-
-    ```bash
-    npm install @rollup/plugin-yaml --save-dev
-    ```
+    <PackageManagerTabs>
```
Comment (sarah11918) [inline src/content/docs/en/recipes/add-yaml-support.mdx:14]: This change is not related to the PR description and is inappropriate in this PR. Please remove and make a separate PR. (Note: I had to spend extra time figuring out this change because there was no original indentation issue in docs.)

File: src/content/docs/en/guides/content-collections.mdx [hunk 3]
```diff
@@ -144,6 +144,8 @@ To define your first collection, create a `src/content/config.ts` file if one do
 2. **Define each collection that you'd like to validate.** This includes a `type` (introduced in Astro v2.5.0) specifying whether the collection contains content authoring formats like Markdown (`type: 'content'`) or data formats like JSON or YAML (`type: 'data'`). It also includes a `schema` that defines the shape of your frontmatter or entry data.
 3. **Export a single `collections` object** to register your collections.
 
+For example, your `content.ts` file could look something like this:
+
```
Comment (sarah11918) [inline src/content/docs/en/guides/content-collections.mdx:148]: I don't know why this was added? It is not appropriate in this PR.

**Review by sarah11918 — COMMENTED** [review]
Some comments going through below! I appreciate your enthusiasm, but as you'll see from some of my comments, this really went off the rails and it can't happen again.  😄 

It's not just that it took me more than 30 minutes to review (instead of 30 seconds), our translators also use these PRs as the source to update language files. Now, it becomes difficult for *them* to update because so many changes happened at once. 

Thanks for understanding and working with our process that tries to make things as straightforward as possible for everyone! 🙌

File: src/content/docs/en/recipes/add-yaml-support.mdx (hunk 2 above)
Comment (VoxelMC) [reply src/content/docs/en/recipes/add-yaml-support.mdx]: Reverted

File: src/content/docs/en/tutorial/3-components/4.mdx (hunk 1 above)
Comment (VoxelMC) [reply src/content/docs/en/tutorial/3-components/4.mdx:102]: Yeah, this really diffed quite strangely... sorry about that.

File: src/content/docs/en/guides/content-collections.mdx (hunk 3 above)
Comment (VoxelMC) [reply src/content/docs/en/guides/content-collections.mdx]: Reverted

File: src/content/docs/en/guides/deploy/space.mdx [hunk 4]
```diff
@@ -39,17 +39,17 @@ Make the following changes to the `Spacefile` file at the root of your project g
...
-v: 0
-micros:
-  - name: static-astro-in-space
-    src: .
-    engine: static
-    commands:
-      - npm run build
-    serve: dist
-```
+    ```yaml title="Spacefile" {6,8,9}
```
[the remark this answers is not in the document]
Comment (VoxelMC) [reply src/content/docs/en/guides/deploy/space.mdx:42]: It was the indentation of the numbered list above it. If you agree with my idea, i will revert this and open a separate PR to fix it there instead.

File: src/content/docs/en/guides/middleware.mdx [hunk 5]
```diff
@@ -17,26 +17,26 @@ Middleware is available in both SSG and SSR Astro projects.
...
-    export function onRequest ({ locals, request }, next) {
-        // intercept response data from a request
-        // optionally, transform the response by modifying `locals`
-        locals.title = "New title";
-
-        // return a Response or the result of calling `next()`
-        return next();
-    };
-    ```
+```js title="src/middleware.js"
```
[the remark this answers is not in the document]
Comment (VoxelMC) [reply src/content/docs/en/guides/middleware.mdx]: Fixed, and checked that it is actually indented. I guess i fixed the one I changed but didn't change the one I was supposed to...

File: src/content/docs/en/core-concepts/layouts.mdx [hunk 6]
```diff
@@ -82,26 +82,26 @@ A typical layout for Markdown or MDX pages includes:
...
-  <body>
-    <!-- Add other UI components here, like common headers and footers. -->
-    <h1>{frontmatter.title} by {frontmatter.author}</h1>
-    <!-- 2. Rendered HTML will be passed into the default slot. -->
-    <slot />
-    <p>Written on: {frontmatter.date}</p>
-  </body>
-</html>
-```
+    ```astro /(?<!//.*){?frontmatter(?:\\.\w+)?}?/ "<slot />"
```
[the remark this answers is not in the document]
Comment (VoxelMC) [reply src/content/docs/en/core-concepts/layouts.mdx]: Reverted

File: src/content/docs/en/recipes/add-yaml-support.mdx [hunk 7]
```diff
@@ -4,17 +4,17 @@ description: Learn how you can import YAML data by adding a Rollup plugin to you
 i18nReady: true
 type: recipe
 ---
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
```
Comment (sarah11918) [inline src/content/docs/en/recipes/add-yaml-support.mdx:7]:
```suggestion
```
If we're not using tabs then we don't need the import either!
